### PR TITLE
pysrc2cpg: Handling for .import use case 

### PR DESCRIPTION
**Pull request by pandurangpatil**
* Handling of .import use case 
* Unit tests for the given use case

--------
student/urls.py 
--------
```

from django.contrib import admin
from django.urls import path
from django.conf.urls import url

from .import views

urlpatterns = [
    url(r'^allPage/$', views.all_page)
]


```
--------
student/views.py
--------

```
def all_page(request):
	print("All pages")
```

The handling of `.import` wasn't there. This code change will handle the given use case. Added respective unit tests as well.

### CODE REVIEW

File: joern-cli/frontends/pysrc2cpg/src/main/scala/io/joern/pysrc2cpg/PythonTypeRecovery.scala [hunk 1]
```diff
@@ -57,7 +57,16 @@ class RecoverForPythonFile(
   override def visitImport(importCall: Call): Unit = {
     importCall.argument.l match {
       case (path: Literal) :: (funcOrModule: Literal) :: alias =>
-        val calleeNames = extractPossibleCalleeNames(path.code, funcOrModule.code)
+        val tpath = if (path.code == ".") {
+          // TODO: pysrc2cpg does not link files to the correct namespace nodes
+          val root     = cpg.metaData.root.headOption.getOrElse("")
+          val fileName = path.file.name.head.stripPrefix(root)
```
Comment (fabsx00) [inline joern-cli/frontends/pysrc2cpg/src/main/scala/io/joern/pysrc2cpg/PythonTypeRecovery.scala:63]: Use `headOption` here and handle case where `path.file.name` is empty.

File: joern-cli/frontends/pysrc2cpg/src/main/scala/io/joern/pysrc2cpg/PythonTypeRecovery.scala [hunk 2]
```diff
@@ -57,7 +57,16 @@ class RecoverForPythonFile(
   override def visitImport(importCall: Call): Unit = {
     importCall.argument.l match {
       case (path: Literal) :: (funcOrModule: Literal) :: alias =>
-        val calleeNames = extractPossibleCalleeNames(path.code, funcOrModule.code)
+        val tpath = if (path.code == ".") {
```
Comment (fabsx00) [inline joern-cli/frontends/pysrc2cpg/src/main/scala/io/joern/pysrc2cpg/PythonTypeRecovery.scala:60]: What is `tpath` short for? Maybe we can find a more expressive name?

**Review by fabsx00 — APPROVED** [review]

File: joern-cli/frontends/pysrc2cpg/src/test/scala/io/joern/pysrc2cpg/dataflow/DataFlowTests.scala [hunk 3]
```diff
@@ -221,6 +221,81 @@ class DataFlowTests extends PySrc2CpgFixture(withOssDataflow = true) {
...
+        |    url(r'^allPage/$', views.all_page)
+        |]
+        |""".stripMargin
+    val views =
+      """
+        |def all_page(request):
+        |	print("All pages")
+        |""".stripMargin
+    val cpg = code("print('Hello, world!')")
+      .moreCode(controller, "controller/urls.py")
```
Comment (DavidBakerEffendi) [inline joern-cli/frontends/pysrc2cpg/src/test/scala/io/joern/pysrc2cpg/dataflow/DataFlowTests.scala:242]:
```suggestion
      .moreCode(controller, Seq("controller", "urls.py").mkString(File.separator))
```
`File` comes from `java.io.File`

File: joern-cli/frontends/pysrc2cpg/src/main/scala/io/joern/pysrc2cpg/PythonTypeRecovery.scala (hunk 1 above)
Comment (pandurangpatil) [reply joern-cli/frontends/pysrc2cpg/src/main/scala/io/joern/pysrc2cpg/PythonTypeRecovery.scala]: done

File: joern-cli/frontends/pysrc2cpg/src/main/scala/io/joern/pysrc2cpg/PythonTypeRecovery.scala (hunk 2 above)
Comment (pandurangpatil) [reply joern-cli/frontends/pysrc2cpg/src/main/scala/io/joern/pysrc2cpg/PythonTypeRecovery.scala]: updated it to `namespace` which will be more appropriate.

File: joern-cli/frontends/pysrc2cpg/src/test/scala/io/joern/pysrc2cpg/dataflow/DataFlowTests.scala (hunk 3 above)
Comment (pandurangpatil) [reply joern-cli/frontends/pysrc2cpg/src/test/scala/io/joern/pysrc2cpg/dataflow/DataFlowTests.scala]: done

File: joern-cli/frontends/pysrc2cpg/src/main/scala/io/joern/pysrc2cpg/PythonTypeRecovery.scala [hunk 4]
```diff
@@ -57,7 +57,16 @@ class RecoverForPythonFile(
   override def visitImport(importCall: Call): Unit = {
     importCall.argument.l match {
       case (path: Literal) :: (funcOrModule: Literal) :: alias =>
-        val calleeNames = extractPossibleCalleeNames(path.code, funcOrModule.code)
+        val namespace = if (path.code == ".") {
+          // TODO: pysrc2cpg does not link files to the correct namespace nodes
+          val root     = cpg.metaData.root.headOption.getOrElse("")
+          val fileName = path.file.name.headOption.getOrElse("").stripPrefix(root)
+          val sep      = java.io.File.separator
```
Comment (DavidBakerEffendi) [inline joern-cli/frontends/pysrc2cpg/src/main/scala/io/joern/pysrc2cpg/PythonTypeRecovery.scala:64]:
```suggestion
          val sep      = Matcher.quoteReplacement(JFile.separator)
```

File: joern-cli/frontends/pysrc2cpg/src/main/scala/io/joern/pysrc2cpg/PythonTypeRecovery.scala [hunk 5]
```diff
@@ -57,7 +57,16 @@ class RecoverForPythonFile(
...
       case (path: Literal) :: (funcOrModule: Literal) :: alias =>
-        val calleeNames = extractPossibleCalleeNames(path.code, funcOrModule.code)
+        val namespace = if (path.code == ".") {
+          // TODO: pysrc2cpg does not link files to the correct namespace nodes
+          val root     = cpg.metaData.root.headOption.getOrElse("")
+          val fileName = path.file.name.headOption.getOrElse("").stripPrefix(root)
+          val sep      = Matcher.quoteReplacement(JFile.separator)
+          // The below gives us the full path of the relative "."
+          (if (fileName.contains(sep)) fileName.substring(0, fileName.lastIndexOf(sep))
+           else "").replaceAll(sep, ".")
```
[the remark this answers is not in the document]
Comment (pandurangpatil) [reply joern-cli/frontends/pysrc2cpg/src/main/scala/io/joern/pysrc2cpg/PythonTypeRecovery.scala]: done

**Pull request merged**